{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SNV Simulation\n",
    "\n",
    "Simulate various kinds of DNA/RNA mutation scenarios. Data output to `test_input/`.\n",
    "\n",
    "Produces three BAM/MAF combinations:\n",
    "* **Wild-Type:** `sim_wt.sorted.bam` & `sim_wt.sorted.maf`\n",
    "* **Heterozygous:** `sim_het.sorted.bam` & `sim_het.sorted.maf`\n",
    "* **Homozygous:** `sim_hom.sorted.bam` & `sim_hom.sorted.maf`\n",
    "\n",
    "Creates symlinks to simulate eight mutation scenarios:\n",
    "RNAed, T-RNAed, VSE, T-VSE, VSL, T-VSL, LOH (amp & del), and SOM."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create \"wild-type\" reads (no SNVs) using `wgsim`, align with `bwa` (creates a SAM), convert to BAM with `samtools`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[wgsim] seed = 8\n",
      "[wgsim_core] calculating the total length of the reference sequence...\n",
      "[wgsim_core] 1 sequences, total length: 1000\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "+ wgsim -N 1000 -r 0 -S 8 -e 0 test_input/e_coli/NC_008253_1K.fna /tmp/tmp.aKC6DQZCUv /dev/null\n",
      "+ bwa mem -M -t 8 -p test_input/e_coli/NC_008253_1K.fna /tmp/tmp.aKC6DQZCUv\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "ref=\"test_input/e_coli/NC_008253_1K.fna\"\n",
    "log_dir=\"test_input/sim_wt/logs\"\n",
    "output_file=\"test_input/sim_wt/sim_wt.sorted.bam\"\n",
    "\n",
    "fq_file=\"$(mktemp)\"\n",
    "sam_file=\"$(mktemp)\"\n",
    "bam_file=\"$(mktemp)\"\n",
    "\n",
    "mkdir -p \"$(dirname $output_file)\"\n",
    "mkdir -p \"$log_dir\"\n",
    "\n",
    "set -euxo pipefail\n",
    "\n",
    "# Generate 1,000 reads, with a 0% rate of mutations and 0% error rate.\n",
    "# Set seed to a constant for reproducability.\n",
    "# wgsim outputs paired-end reads; we send second ends to /dev/null to get single-end reads.\n",
    "wgsim -N 1000 -r 0 -S 8 -e 0 \"$ref\" \"$fq_file\" /dev/null 2>&1 >\"$log_dir\"/wgsim.log\n",
    "bwa mem -M -t 8 -p \"$ref\" \"$fq_file\" >>\"$sam_file\" 2>\"$log_dir\"/bwa.log\n",
    "samtools view -b -S -o \"$bam_file\" \"$sam_file\"\n",
    "samtools sort \"$bam_file\" -o \"$output_file\"\n",
    "samtools index \"$output_file\"\n",
    "\n",
    "# Clean up intermediate files\n",
    "rm \"$fq_file\"\n",
    "rm \"$sam_file\"\n",
    "rm \"$bam_file\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Produce simulated mutated reads using `bamsurgeon`. We just need three types of mutated BAMs to construct all simulated allelic asymmetries:\n",
    "* Homozygous ref (already created above).\n",
    "* Heterozygous (50% ref, 50% alt)\n",
    "* Homozygous alt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[Fri Apr 14 19:12:23 EDT 2017] picard.sam.SamToFastq INPUT=test_input/sim_het/logs/haplo_gi|110640213|ref|NC_008253.1|_200_200.tmpbam.2ddc01a1-2da2-4ab7-acfd-814e7abdf287.bam FASTQ=test_input/sim_het/logs/haplo_gi|110640213|ref|NC_008253.1|_200_200.tmpbam.2ddc01a1-2da2-4ab7-acfd-814e7abdf287.fastq INCLUDE_NON_PRIMARY_ALIGNMENTS=false VALIDATION_STRINGENCY=SILENT    OUTPUT_PER_RG=false RG_TAG=PU RE_REVERSE=true INTERLEAVE=false INCLUDE_NON_PF_READS=false CLIPPING_MIN_LENGTH=0 READ1_TRIM=0 READ2_TRIM=0 VERBOSITY=INFO QUIET=false COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json\n",
      "[Fri Apr 14 19:12:23 EDT 2017] Executing as moorena@cga02.broadinstitute.org on Linux 2.6.32-642.15.1.el6.x86_64 amd64; OpenJDK 64-Bit Server VM 1.8.0_92-b15; Picard version: 2.9.0-SNAPSHOT\n",
      "[Fri Apr 14 19:12:23 EDT 2017] picard.sam.SamToFastq done. Elapsed time: 0.00 minutes.\n",
      "Runtime.totalMemory()=2058354688\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 74 sequences (5180 bp)...\n",
      "[M::mem_process_seqs] Processed 74 reads in 0.003 CPU sec, 0.003 real sec\n",
      "[main] Version: 0.7.12-r1039\n",
      "[main] CMD: bwa mem -t 1 -M test_input/e_coli/NC_008253_1K.fna test_input/sim_het/logs/haplo_gi|110640213|ref|NC_008253.1|_200_200.tmpbam.2ddc01a1-2da2-4ab7-acfd-814e7abdf287.fastq\n",
      "[main] Real time: 0.009 sec; CPU: 0.007 sec\n",
      "[Fri Apr 14 19:12:26 EDT 2017] picard.sam.SamToFastq INPUT=test_input/sim_hom/logs/haplo_gi|110640213|ref|NC_008253.1|_200_200.tmpbam.b5b06b2a-157b-48ef-a05e-f68243ab58b3.bam FASTQ=test_input/sim_hom/logs/haplo_gi|110640213|ref|NC_008253.1|_200_200.tmpbam.b5b06b2a-157b-48ef-a05e-f68243ab58b3.fastq INCLUDE_NON_PRIMARY_ALIGNMENTS=false VALIDATION_STRINGENCY=SILENT    OUTPUT_PER_RG=false RG_TAG=PU RE_REVERSE=true INTERLEAVE=false INCLUDE_NON_PF_READS=false CLIPPING_MIN_LENGTH=0 READ1_TRIM=0 READ2_TRIM=0 VERBOSITY=INFO QUIET=false COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json\n",
      "[Fri Apr 14 19:12:26 EDT 2017] Executing as moorena@cga02.broadinstitute.org on Linux 2.6.32-642.15.1.el6.x86_64 amd64; OpenJDK 64-Bit Server VM 1.8.0_92-b15; Picard version: 2.9.0-SNAPSHOT\n",
      "[Fri Apr 14 19:12:26 EDT 2017] picard.sam.SamToFastq done. Elapsed time: 0.00 minutes.\n",
      "Runtime.totalMemory()=2058354688\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 74 sequences (5180 bp)...\n",
      "[M::mem_process_seqs] Processed 74 reads in 0.003 CPU sec, 0.003 real sec\n",
      "[main] Version: 0.7.12-r1039\n",
      "[main] CMD: bwa mem -t 1 -M test_input/e_coli/NC_008253_1K.fna test_input/sim_hom/logs/haplo_gi|110640213|ref|NC_008253.1|_200_200.tmpbam.b5b06b2a-157b-48ef-a05e-f68243ab58b3.fastq\n",
      "[main] Real time: 0.007 sec; CPU: 0.007 sec\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "++ mktemp\n",
      "+ spikein_file=/tmp/tmp.6dijihv4EW\n",
      "+ echo 'gi|110640213|ref|NC_008253.1|   200     200     0.5     C'\n",
      "+ bamsurgeon addsnv.py --single --picardjar /seq/software/picard/current/bin/picard.jar --aligner mem -v /tmp/tmp.6dijihv4EW -f test_input/sim_wt/sim_wt.sorted.bam -r test_input/e_coli/NC_008253_1K.fna -o /tmp/tmp.xL89jSyUqY --tmpdir test_input/sim_het/logs\n",
      "+ samtools sort /tmp/tmp.xL89jSyUqY -o test_input/sim_het/sim_het.sorted.bam\n",
      "+ samtools index test_input/sim_het/sim_het.sorted.bam\n",
      "+ rm -r addsnv_logs_tmp.xL89jSyUqY\n",
      "+ echo 'gi|110640213|ref|NC_008253.1|   200     200     1     C'\n",
      "+ bamsurgeon addsnv.py --single --picardjar /seq/software/picard/current/bin/picard.jar --aligner mem -v /tmp/tmp.6dijihv4EW -f test_input/sim_wt/sim_wt.sorted.bam -r test_input/e_coli/NC_008253_1K.fna -o /tmp/tmp.HNilmEjOrV --tmpdir test_input/sim_hom/logs\n",
      "+ samtools sort /tmp/tmp.HNilmEjOrV -o test_input/sim_hom/sim_hom.sorted.bam\n",
      "+ samtools index test_input/sim_hom/sim_hom.sorted.bam\n",
      "+ rm -r addsnv_logs_tmp.HNilmEjOrV\n",
      "+ rm /tmp/tmp.xL89jSyUqY\n",
      "+ rm /tmp/tmp.HNilmEjOrV\n",
      "+ rm /tmp/tmp.6dijihv4EW\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "ref=\"test_input/e_coli/NC_008253_1K.fna\"\n",
    "picard_jar=\"/seq/software/picard/current/bin/picard.jar\"\n",
    "sim_wt_bam=\"test_input/sim_wt/sim_wt.sorted.bam\"\n",
    "het_output=\"test_input/sim_het/sim_het.sorted.bam\"\n",
    "hom_output=\"test_input/sim_hom/sim_hom.sorted.bam\"\n",
    "het_logs=\"test_input/sim_het/logs\"\n",
    "hom_logs=\"test_input/sim_hom/logs\"\n",
    "\n",
    "chrom=\"gi|110640213|ref|NC_008253.1|\"\n",
    "snv_base=\"C\"\n",
    "snv_pos=\"200\"\n",
    "het_vaf=\"0.5\"\n",
    "hom_vaf=\"1\"\n",
    "\n",
    "mkdir -p \"$(dirname $het_output)\"\n",
    "mkdir -p \"$(dirname $hom_output)\"\n",
    "mkdir -p \"$het_logs\"\n",
    "mkdir -p \"$hom_logs\"\n",
    "het_bam_file=\"$(mktemp)\"\n",
    "hom_bam_file=\"$(mktemp)\"\n",
    "\n",
    "set -euxo pipefail\n",
    "\n",
    "# Make heterozygous BAM\n",
    "spikein_file=\"$(mktemp)\"\n",
    "echo \"$chrom   $snv_pos     $snv_pos     $het_vaf     $snv_base\" >\"$spikein_file\"\n",
    "\n",
    "bamsurgeon addsnv.py \\\n",
    "    --single \\\n",
    "    --picardjar \"$picard_jar\" \\\n",
    "    --aligner mem \\\n",
    "    -v \"$spikein_file\" \\\n",
    "    -f \"$sim_wt_bam\" \\\n",
    "    -r \"$ref\" \\\n",
    "    -o \"$het_bam_file\" \\\n",
    "    --tmpdir \"$het_logs\" \\\n",
    "    2>&1 >\"$het_logs/bamsurgeon.log\"\n",
    "\n",
    "samtools sort \"$het_bam_file\" -o \"$het_output\"\n",
    "samtools index \"$het_output\"\n",
    "rm -r \"addsnv_logs_tmp.\"*\n",
    "\n",
    "# Make homozygous BAM\n",
    "echo \"$chrom   $snv_pos     $snv_pos     $hom_vaf     $snv_base\" >\"$spikein_file\"\n",
    "\n",
    "bamsurgeon addsnv.py \\\n",
    "    --single \\\n",
    "    --picardjar \"$picard_jar\" \\\n",
    "    --aligner mem \\\n",
    "    -v \"$spikein_file\" \\\n",
    "    -f \"$sim_wt_bam\" \\\n",
    "    -r \"$ref\" \\\n",
    "    -o \"$hom_bam_file\" \\\n",
    "    --tmpdir \"$hom_logs\" \\\n",
    "    2>&1 >\"$hom_logs/bamsurgeon.log\"\n",
    "\n",
    "samtools sort \"$hom_bam_file\" -o \"$hom_output\"\n",
    "samtools index \"$hom_output\"\n",
    "rm -r \"addsnv_logs_tmp.\"*\n",
    "\n",
    "rm \"$het_bam_file\"\n",
    "rm \"$hom_bam_file\"\n",
    "rm \"$spikein_file\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " Call variants using `samtools mpileup` and `bcftools`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "+ for type in '\"${!bams[@]}\"'\n",
      "+ samtools mpileup -g -f test_input/e_coli/NC_008253_1K.fna test_input/sim_het/sim_het.sorted.bam\n",
      "+ bcftools call -c -v -\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n",
      "[mpileup] 1 samples in 1 input files\n",
      "<mpileup> Set max per-file depth to 8000\n",
      "+ for type in '\"${!bams[@]}\"'\n",
      "+ samtools mpileup -g -f test_input/e_coli/NC_008253_1K.fna test_input/sim_hom/sim_hom.sorted.bam\n",
      "+ bcftools call -c -v -\n",
      "[mpileup] 1 samples in 1 input files\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n",
      "<mpileup> Set max per-file depth to 8000\n",
      "+ for type in '\"${!bams[@]}\"'\n",
      "+ samtools mpileup -g -f test_input/e_coli/NC_008253_1K.fna test_input/sim_wt/sim_wt.sorted.bam\n",
      "+ bcftools call -c -v -\n",
      "[mpileup] 1 samples in 1 input files\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n",
      "<mpileup> Set max per-file depth to 8000\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "declare -A bams=(\n",
    "    [\"wt\"]=\"test_input/sim_wt/sim_wt.sorted.bam\" \\\n",
    "    [\"het\"]=\"test_input/sim_het/sim_het.sorted.bam\" \\\n",
    "    [\"hom\"]=\"test_input/sim_hom/sim_hom.sorted.bam\" \\\n",
    ")\n",
    "\n",
    "declare -A outputs=(\n",
    "    [\"wt\"]=\"test_input/sim_wt/sim_wt.sorted.vcf\" \\\n",
    "    [\"het\"]=\"test_input/sim_het/sim_het.sorted.vcf\" \\\n",
    "    [\"hom\"]=\"test_input/sim_hom/sim_hom.sorted.vcf\" \\\n",
    ")\n",
    "\n",
    "ref=\"test_input/e_coli/NC_008253_1K.fna\"\n",
    "\n",
    "set -euxo pipefail\n",
    "\n",
    "for type in \"${!bams[@]}\"; do\n",
    "    samtools mpileup -g -f \"$ref\" \"${bams[$type]}\" | \\\n",
    "        bcftools call -c -v - >\"${outputs[$type]}\"\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Verbose mode on\n",
      "Path:\n",
      "['/xchip/tcga/Tools/oncotator/onco_env_2.7.1/bin', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/site-packages/Oncotator-1.9.0.0-py2.7.egg', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python27.zip', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/plat-linux2', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/lib-tk', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/lib-old', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/lib-dynload', '/broad/software/free/Linux/redhat_6_x86_64/pkgs/python_2.7.1-sqlite3-rtrees/lib/python2.7', '/broad/software/free/Linux/redhat_6_x86_64/pkgs/python_2.7.1-sqlite3-rtrees/lib/python2.7/plat-linux2', '/broad/software/free/Linux/redhat_6_x86_64/pkgs/python_2.7.1-sqlite3-rtrees/lib/python2.7/lib-tk', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/site-packages']\n",
      " \n",
      "Verbose mode on\n",
      "Path:\n",
      "['/xchip/tcga/Tools/oncotator/onco_env_2.7.1/bin', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/site-packages/Oncotator-1.9.0.0-py2.7.egg', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python27.zip', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/plat-linux2', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/lib-tk', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/lib-old', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/lib-dynload', '/broad/software/free/Linux/redhat_6_x86_64/pkgs/python_2.7.1-sqlite3-rtrees/lib/python2.7', '/broad/software/free/Linux/redhat_6_x86_64/pkgs/python_2.7.1-sqlite3-rtrees/lib/python2.7/plat-linux2', '/broad/software/free/Linux/redhat_6_x86_64/pkgs/python_2.7.1-sqlite3-rtrees/lib/python2.7/lib-tk', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/site-packages']\n",
      " \n",
      "Verbose mode on\n",
      "Path:\n",
      "['/xchip/tcga/Tools/oncotator/onco_env_2.7.1/bin', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/site-packages/Oncotator-1.9.0.0-py2.7.egg', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python27.zip', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/plat-linux2', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/lib-tk', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/lib-old', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/lib-dynload', '/broad/software/free/Linux/redhat_6_x86_64/pkgs/python_2.7.1-sqlite3-rtrees/lib/python2.7', '/broad/software/free/Linux/redhat_6_x86_64/pkgs/python_2.7.1-sqlite3-rtrees/lib/python2.7/plat-linux2', '/broad/software/free/Linux/redhat_6_x86_64/pkgs/python_2.7.1-sqlite3-rtrees/lib/python2.7/lib-tk', '/xchip/tcga/Tools/oncotator/onco_env_2.7.1/lib/python2.7/site-packages']\n",
      " \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "+ for type in '\"${!vcfs[@]}\"'\n",
      "++ dirname test_input/sim_het/sim_het.sorted.vcf\n",
      "+ log_loc=test_input/sim_het/logs/oncotator.log\n",
      "+ oncotator --input_format VCF --log_name test_input/sim_het/logs/oncotator.log test_input/sim_het/sim_het.sorted.vcf test_input/sim_het/sim_het.sorted.maf hg19\n",
      "2017-04-14 19:13:02,142 INFO [oncotator.Oncotator:239] Oncotator v1.9.0.0\n",
      "2017-04-14 19:13:02,142 INFO [oncotator.Oncotator:240] Args: Namespace(allow_overwriting=False, cache_url=None, canonical_tx_file=None, collapse_filter_cols=False, collapse_number_annotations=False, dbDir='/xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/', default_cli=[], default_config=None, genome_build='hg19', infer_genotypes='false', infer_onps=False, input_file='test_input/sim_het/sim_het.sorted.vcf', input_format='VCF', log_name='test_input/sim_het/logs/oncotator.log', noMulticore=False, output_file='test_input/sim_het/sim_het.sorted.maf', output_format='TCGAMAF', override_cli=[], override_config=None, prepend=False, read_only_cache=False, reannotate_tcga_maf_cols=False, skip_no_alt=False, tx_mode='CANONICAL', verbose=5)\n",
      "2017-04-14 19:13:02,143 INFO [oncotator.Oncotator:241] Log file: /xchip/cga_home/moorena/projects/r2d2/test_input/sim_het/logs/oncotator.log\n",
      "2017-04-14 19:13:02,456 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cosmic_fusion/hg19/cosmic_fusion.config\n",
      "2017-04-14 19:13:02,457 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/gencode_xrefseq/hg19/gencode_xrefseq.config\n",
      "2017-04-14 19:13:02,458 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/clinvar/hg19/clinvar.config\n",
      "2017-04-14 19:13:02,459 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/esp6500SI_v2/hg19/esp6500SI_v2.config\n",
      "2017-04-14 19:13:02,460 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ccle_by_gene/hg19/ccle_by_gene.config\n",
      "2017-04-14 19:13:02,461 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/oreganno/hg19/oreganno.config\n",
      "2017-04-14 19:13:02,461 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/achilles/hg19/achilles.config\n",
      "2017-04-14 19:13:02,462 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cancer_gene_census/hg19/cancer_gene_census.config\n",
      "2017-04-14 19:13:02,463 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/simple_uniprot/hg19/simple_uniprot.config\n",
      "2017-04-14 19:13:02,464 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/dna_repair_genes/hg19/dna_repair_genes.config\n",
      "2017-04-14 19:13:02,465 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/dbsnp/hg19/dbsnp.config\n",
      "2017-04-14 19:13:02,465 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ref_hg/hg19/ref_hg.config\n",
      "2017-04-14 19:13:02,466 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/uniprot_aa_xform/hg19/uniprot_aa_xform.config\n",
      "2017-04-14 19:13:02,467 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/gencode_out2/hg19/gencode_out2.config\n",
      "2017-04-14 19:13:02,467 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/hgnc/hg19/hgnc.config\n",
      "2017-04-14 19:13:02,468 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ccle_by_gp/hg19/ccle_by_gp.config\n",
      "2017-04-14 19:13:02,469 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cosmic_tissue/hg19/cosmic_tissue.config\n",
      "2017-04-14 19:13:02,470 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/familial/hg19/familial.config\n",
      "2017-04-14 19:13:02,471 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/tumorscape/hg19/tumorscape.config\n",
      "2017-04-14 19:13:02,472 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cosmic/hg19/cosmic.config\n",
      "2017-04-14 19:13:02,472 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/1000genome_db/hg19/1000genome_db.config\n",
      "2017-04-14 19:13:02,473 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/uniprot_aa_annotation/hg19/uniprot_aa_annotation.config\n",
      "2017-04-14 19:13:02,474 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/dbNSFP_ds/hg19/dbNSFP_ds.config\n",
      "2017-04-14 19:13:02,475 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/esp6500SI_v2_coverage/hg19/esp6500SI_v2_coverage.config\n",
      "2017-04-14 19:13:02,476 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/so_terms/hg19/so_terms.config\n",
      "2017-04-14 19:13:02,477 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/tcgascape/hg19/tcgascape.config\n",
      "2017-04-14 19:13:02,478 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/mutsig/hg19/mutsig.config\n",
      "2017-04-14 19:13:02,478 INFO [oncotator.DatasourceFactory:304] Creating pool\n",
      "2017-04-14 19:13:02,492 INFO [oncotator.DatasourceFactory:307] Pool created\n",
      "2017-04-14 19:13:02,599 INFO [oncotator.datasources.EnsemblTranscriptDatasource:95] GENCODE v19 is being set up in faster, NOT thread-safe mode (for annotation).  \n",
      "2017-04-14 19:13:02,684 INFO [oncotator.datasources.EnsemblTranscriptDatasource:106] GENCODE v19 is being set up with default tx-mode: CANONICAL.  \n",
      "2017-04-14 19:13:02,684 WARNING [oncotator.datasources.EnsemblTranscriptDatasource:123] Attempting to set transcript mode of CANONICAL for ensembl.  This operation is only supported for GENCODE.  Otherwise, will be the same as EFFECT.\n",
      "2017-04-14 19:13:02,685 INFO [oncotator.datasources.EnsemblTranscriptDatasource:109] GENCODE v19 is being set up with basic filtering.  \n",
      "2017-04-14 19:13:02,906 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/so_terms/hg19//SO_Ensembl_consequence_terms.bin...\n",
      "2017-04-14 19:13:02,919 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/oreganno/hg19//oreganno_results_by_pos.hg19.import.bin...\n",
      "2017-04-14 19:13:02,919 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cancer_gene_census/hg19//CancerGeneCensus_Table_1_full_2012-03-15.bin...\n",
      "2017-04-14 19:13:02,919 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/clinvar/hg19//clinvar_hgmd.12-03-20.bin...\n",
      "2017-04-14 19:13:02,919 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cosmic_fusion/hg19//CosmicMutantExportIncFus_v62_291112_fusion.bin...\n",
      "2017-04-14 19:13:02,925 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/gencode_xrefseq/hg19//gencode_xrefseq_v19.bin...\n",
      "2017-04-14 19:13:02,935 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/simple_uniprot/hg19//simple_uniprot_Dec012014.bin...\n",
      "2017-04-14 19:13:03,484 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/dna_repair_genes/hg19//Wood_DNARepairGenes.no_dupes.bin...\n",
      "2017-04-14 19:13:03,490 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/hgnc/hg19//hgnc_download_Sept172014.bin...\n",
      "2017-04-14 19:13:03,557 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/achilles/hg19//achilles_lineage_results.import.bin...\n",
      "2017-04-14 19:13:03,699 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ccle_by_gp/hg19//ccle_results_by_pos.hg19.import.bin...\n",
      "2017-04-14 19:13:03,712 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cosmic_tissue/hg19//CosmicMutantExportIncFus_v62_291112.geneHistology.bin...\n",
      "2017-04-14 19:13:03,870 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/familial/hg19//Familial Cancer Genes.no_dupes.bin...\n",
      "2017-04-14 19:13:03,880 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/tumorscape/hg19//tumorscape_20100104_generated.bin...\n",
      "2017-04-14 19:13:04,040 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/uniprot_aa_annotation/hg19//uniprot_aa_annotations.bin...\n",
      "2017-04-14 19:13:04,101 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/mutsig/hg19//mutsig_results.import.20110905.bin...\n",
      "2017-04-14 19:13:05,172 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ccle_by_gene/hg19//ccle_results_by_gene.import.bin...\n",
      "2017-04-14 19:13:08,253 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/tcgascape/hg19//tcgascape_110405_generated.bin...\n",
      "2017-04-14 19:13:13,018 INFO [oncotator.DatasourceFactory:321] Mapping complete: 17 datasources created in multiprocess\n",
      "2017-04-14 19:13:13,299 INFO [oncotator.utils.RunSpecificationFactory:124] Setting GENCODE v19 to tx-mode of CANONICAL...\n",
      "2017-04-14 19:13:13,299 WARNING [oncotator.datasources.EnsemblTranscriptDatasource:123] Attempting to set transcript mode of CANONICAL for ensembl.  This operation is only supported for GENCODE.  Otherwise, will be the same as EFFECT.\n",
      "2017-04-14 19:13:13,299 INFO [oncotator.utils.RunSpecificationFactory:135] No custom canonical transcripts specified.\n",
      "2017-04-14 19:13:13,302 INFO [oncotator.utils.ConfigUtils:197] Could not find config file (tcgaMAF2.4_output.config).  Trying configs/ prepend.\n",
      "2017-04-14 19:13:13,311 INFO [oncotator.output.TcgaMafOutputRenderer:93] Building alternative keys dictionary...\n",
      "2017-04-14 19:13:13,345 INFO [oncotator.cache.DummyCache:57] No cache specified.  All cache attempts will be listed as cache misses.\n",
      "2017-04-14 19:13:13,345 INFO [oncotator.Annotator:426] Annotating with 27 datasources: Oncotator v1.9.0.0 | Flat File Reference hg19 | GENCODE v19 CANONICAL | UniProt_AAxform 2014_12 | ESP 6500SI-V2 | dbSNP build 134 | COSMIC v62_291112 | 1000Genome phase1 | dbNSFP v2.4 | ESP 6500SI-V2 | ClinVar 12.03.20 | ORegAnno UCSC Track | CCLE_By_GP 09292010 | UniProt_AA 2014_12 | Ensembl ICGC MUCOPA | COSMIC_FusionGenes v62_291112 | gencode_xref_refseq metadata_v19 | CCLE_By_Gene 09292010 | ACHILLES_Lineage_Results 110303 | CGC full_2012-03-15 | UniProt 2014_12 | HumanDNARepairGenes 20110905 | HGNC Sept172014 | COSMIC_Tissue 291112 | Familial_Cancer_Genes 20110905 | TUMORScape 20100104 | TCGAScape 110405 | MutSig Published Results 20110905 \n",
      "2017-04-14 19:13:13,346 INFO [oncotator.utils.ConfigUtils:197] Could not find config file (vcf.in.config).  Trying configs/ prepend.\n",
      "2017-04-14 19:13:13,364 INFO [oncotator.output.TcgaMafOutputRenderer:256] TCGA MAF output file: test_input/sim_het/sim_het.sorted.maf\n",
      "2017-04-14 19:13:13,364 INFO [oncotator.output.TcgaMafOutputRenderer:257] Render starting...\n",
      "2017-04-14 19:13:13,365 INFO [oncotator.utils.ConfigUtils:197] Could not find config file (vcf.in.config).  Trying configs/ prepend.\n",
      "2017-04-14 19:13:13,385 WARNING [oncotator.datasources.ReferenceDatasource:123] /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ref_hg/hg19///chrgi|110640213|ref|NC_008253.1|.txt not found.  Please add it.\n",
      "2017-04-14 19:13:13,385 WARNING [oncotator.datasources.ReferenceDatasource:123] /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ref_hg/hg19///chrgi|110640213|ref|NC_008253.1|.txt not found.  Please add it.\n",
      "2017-04-14 19:13:13,689 INFO [oncotator.output.TcgaMafOutputRenderer:342] Rendered all 1 mutations.\n",
      "+ for type in '\"${!vcfs[@]}\"'\n",
      "++ dirname test_input/sim_hom/sim_hom.sorted.vcf\n",
      "+ log_loc=test_input/sim_hom/logs/oncotator.log\n",
      "+ oncotator --input_format VCF --log_name test_input/sim_hom/logs/oncotator.log test_input/sim_hom/sim_hom.sorted.vcf test_input/sim_hom/sim_hom.sorted.maf hg19\n",
      "2017-04-14 19:13:15,459 INFO [oncotator.Oncotator:239] Oncotator v1.9.0.0\n",
      "2017-04-14 19:13:15,459 INFO [oncotator.Oncotator:240] Args: Namespace(allow_overwriting=False, cache_url=None, canonical_tx_file=None, collapse_filter_cols=False, collapse_number_annotations=False, dbDir='/xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/', default_cli=[], default_config=None, genome_build='hg19', infer_genotypes='false', infer_onps=False, input_file='test_input/sim_hom/sim_hom.sorted.vcf', input_format='VCF', log_name='test_input/sim_hom/logs/oncotator.log', noMulticore=False, output_file='test_input/sim_hom/sim_hom.sorted.maf', output_format='TCGAMAF', override_cli=[], override_config=None, prepend=False, read_only_cache=False, reannotate_tcga_maf_cols=False, skip_no_alt=False, tx_mode='CANONICAL', verbose=5)\n",
      "2017-04-14 19:13:15,459 INFO [oncotator.Oncotator:241] Log file: /xchip/cga_home/moorena/projects/r2d2/test_input/sim_hom/logs/oncotator.log\n",
      "2017-04-14 19:13:15,462 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cosmic_fusion/hg19/cosmic_fusion.config\n",
      "2017-04-14 19:13:15,462 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/gencode_xrefseq/hg19/gencode_xrefseq.config\n",
      "2017-04-14 19:13:15,462 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/clinvar/hg19/clinvar.config\n",
      "2017-04-14 19:13:15,462 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/esp6500SI_v2/hg19/esp6500SI_v2.config\n",
      "2017-04-14 19:13:15,463 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ccle_by_gene/hg19/ccle_by_gene.config\n",
      "2017-04-14 19:13:15,463 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/oreganno/hg19/oreganno.config\n",
      "2017-04-14 19:13:15,463 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/achilles/hg19/achilles.config\n",
      "2017-04-14 19:13:15,463 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cancer_gene_census/hg19/cancer_gene_census.config\n",
      "2017-04-14 19:13:15,464 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/simple_uniprot/hg19/simple_uniprot.config\n",
      "2017-04-14 19:13:15,464 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/dna_repair_genes/hg19/dna_repair_genes.config\n",
      "2017-04-14 19:13:15,464 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/dbsnp/hg19/dbsnp.config\n",
      "2017-04-14 19:13:15,464 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ref_hg/hg19/ref_hg.config\n",
      "2017-04-14 19:13:15,465 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/uniprot_aa_xform/hg19/uniprot_aa_xform.config\n",
      "2017-04-14 19:13:15,465 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/gencode_out2/hg19/gencode_out2.config\n",
      "2017-04-14 19:13:15,465 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/hgnc/hg19/hgnc.config\n",
      "2017-04-14 19:13:15,465 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ccle_by_gp/hg19/ccle_by_gp.config\n",
      "2017-04-14 19:13:15,465 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cosmic_tissue/hg19/cosmic_tissue.config\n",
      "2017-04-14 19:13:15,466 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/familial/hg19/familial.config\n",
      "2017-04-14 19:13:15,466 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/tumorscape/hg19/tumorscape.config\n",
      "2017-04-14 19:13:15,466 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cosmic/hg19/cosmic.config\n",
      "2017-04-14 19:13:15,466 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/1000genome_db/hg19/1000genome_db.config\n",
      "2017-04-14 19:13:15,467 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/uniprot_aa_annotation/hg19/uniprot_aa_annotation.config\n",
      "2017-04-14 19:13:15,467 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/dbNSFP_ds/hg19/dbNSFP_ds.config\n",
      "2017-04-14 19:13:15,467 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/esp6500SI_v2_coverage/hg19/esp6500SI_v2_coverage.config\n",
      "2017-04-14 19:13:15,467 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/so_terms/hg19/so_terms.config\n",
      "2017-04-14 19:13:15,468 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/tcgascape/hg19/tcgascape.config\n",
      "2017-04-14 19:13:15,468 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/mutsig/hg19/mutsig.config\n",
      "2017-04-14 19:13:15,468 INFO [oncotator.DatasourceFactory:304] Creating pool\n",
      "2017-04-14 19:13:15,479 INFO [oncotator.DatasourceFactory:307] Pool created\n",
      "2017-04-14 19:13:15,563 INFO [oncotator.datasources.EnsemblTranscriptDatasource:95] GENCODE v19 is being set up in faster, NOT thread-safe mode (for annotation).  \n",
      "2017-04-14 19:13:15,645 INFO [oncotator.datasources.EnsemblTranscriptDatasource:106] GENCODE v19 is being set up with default tx-mode: CANONICAL.  \n",
      "2017-04-14 19:13:15,646 WARNING [oncotator.datasources.EnsemblTranscriptDatasource:123] Attempting to set transcript mode of CANONICAL for ensembl.  This operation is only supported for GENCODE.  Otherwise, will be the same as EFFECT.\n",
      "2017-04-14 19:13:15,646 INFO [oncotator.datasources.EnsemblTranscriptDatasource:109] GENCODE v19 is being set up with basic filtering.  \n",
      "2017-04-14 19:13:15,843 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/so_terms/hg19//SO_Ensembl_consequence_terms.bin...\n",
      "2017-04-14 19:13:15,855 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/oreganno/hg19//oreganno_results_by_pos.hg19.import.bin...\n",
      "2017-04-14 19:13:15,855 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cancer_gene_census/hg19//CancerGeneCensus_Table_1_full_2012-03-15.bin...\n",
      "2017-04-14 19:13:15,855 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cosmic_fusion/hg19//CosmicMutantExportIncFus_v62_291112_fusion.bin...\n",
      "2017-04-14 19:13:15,855 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/clinvar/hg19//clinvar_hgmd.12-03-20.bin...\n",
      "2017-04-14 19:13:15,859 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/gencode_xrefseq/hg19//gencode_xrefseq_v19.bin...\n",
      "2017-04-14 19:13:15,869 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/simple_uniprot/hg19//simple_uniprot_Dec012014.bin...\n",
      "2017-04-14 19:13:16,414 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/dna_repair_genes/hg19//Wood_DNARepairGenes.no_dupes.bin...\n",
      "2017-04-14 19:13:16,418 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/hgnc/hg19//hgnc_download_Sept172014.bin...\n",
      "2017-04-14 19:13:16,472 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/achilles/hg19//achilles_lineage_results.import.bin...\n",
      "2017-04-14 19:13:16,612 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ccle_by_gp/hg19//ccle_results_by_pos.hg19.import.bin...\n",
      "2017-04-14 19:13:16,623 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cosmic_tissue/hg19//CosmicMutantExportIncFus_v62_291112.geneHistology.bin...\n",
      "2017-04-14 19:13:16,773 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/familial/hg19//Familial Cancer Genes.no_dupes.bin...\n",
      "2017-04-14 19:13:16,779 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/tumorscape/hg19//tumorscape_20100104_generated.bin...\n",
      "2017-04-14 19:13:16,938 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/uniprot_aa_annotation/hg19//uniprot_aa_annotations.bin...\n",
      "2017-04-14 19:13:16,996 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/mutsig/hg19//mutsig_results.import.20110905.bin...\n",
      "2017-04-14 19:13:17,837 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ccle_by_gene/hg19//ccle_results_by_gene.import.bin...\n",
      "2017-04-14 19:13:21,069 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/tcgascape/hg19//tcgascape_110405_generated.bin...\n",
      "2017-04-14 19:13:25,815 INFO [oncotator.DatasourceFactory:321] Mapping complete: 17 datasources created in multiprocess\n",
      "2017-04-14 19:13:26,098 INFO [oncotator.utils.RunSpecificationFactory:124] Setting GENCODE v19 to tx-mode of CANONICAL...\n",
      "2017-04-14 19:13:26,099 WARNING [oncotator.datasources.EnsemblTranscriptDatasource:123] Attempting to set transcript mode of CANONICAL for ensembl.  This operation is only supported for GENCODE.  Otherwise, will be the same as EFFECT.\n",
      "2017-04-14 19:13:26,099 INFO [oncotator.utils.RunSpecificationFactory:135] No custom canonical transcripts specified.\n",
      "2017-04-14 19:13:26,102 INFO [oncotator.utils.ConfigUtils:197] Could not find config file (tcgaMAF2.4_output.config).  Trying configs/ prepend.\n",
      "2017-04-14 19:13:26,110 INFO [oncotator.output.TcgaMafOutputRenderer:93] Building alternative keys dictionary...\n",
      "2017-04-14 19:13:26,144 INFO [oncotator.cache.DummyCache:57] No cache specified.  All cache attempts will be listed as cache misses.\n",
      "2017-04-14 19:13:26,144 INFO [oncotator.Annotator:426] Annotating with 27 datasources: Oncotator v1.9.0.0 | Flat File Reference hg19 | GENCODE v19 CANONICAL | UniProt_AAxform 2014_12 | ESP 6500SI-V2 | dbSNP build 134 | COSMIC v62_291112 | 1000Genome phase1 | dbNSFP v2.4 | ESP 6500SI-V2 | ClinVar 12.03.20 | ORegAnno UCSC Track | CCLE_By_GP 09292010 | UniProt_AA 2014_12 | Ensembl ICGC MUCOPA | COSMIC_FusionGenes v62_291112 | gencode_xref_refseq metadata_v19 | CCLE_By_Gene 09292010 | ACHILLES_Lineage_Results 110303 | CGC full_2012-03-15 | UniProt 2014_12 | HumanDNARepairGenes 20110905 | HGNC Sept172014 | COSMIC_Tissue 291112 | Familial_Cancer_Genes 20110905 | TUMORScape 20100104 | TCGAScape 110405 | MutSig Published Results 20110905 \n",
      "2017-04-14 19:13:26,144 INFO [oncotator.utils.ConfigUtils:197] Could not find config file (vcf.in.config).  Trying configs/ prepend.\n",
      "2017-04-14 19:13:26,163 INFO [oncotator.output.TcgaMafOutputRenderer:256] TCGA MAF output file: test_input/sim_hom/sim_hom.sorted.maf\n",
      "2017-04-14 19:13:26,163 INFO [oncotator.output.TcgaMafOutputRenderer:257] Render starting...\n",
      "2017-04-14 19:13:26,163 INFO [oncotator.utils.ConfigUtils:197] Could not find config file (vcf.in.config).  Trying configs/ prepend.\n",
      "2017-04-14 19:13:26,183 WARNING [oncotator.datasources.ReferenceDatasource:123] /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ref_hg/hg19///chrgi|110640213|ref|NC_008253.1|.txt not found.  Please add it.\n",
      "2017-04-14 19:13:26,183 WARNING [oncotator.datasources.ReferenceDatasource:123] /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ref_hg/hg19///chrgi|110640213|ref|NC_008253.1|.txt not found.  Please add it.\n",
      "2017-04-14 19:13:26,486 INFO [oncotator.output.TcgaMafOutputRenderer:342] Rendered all 1 mutations.\n",
      "+ for type in '\"${!vcfs[@]}\"'\n",
      "++ dirname test_input/sim_wt/sim_wt.sorted.vcf\n",
      "+ log_loc=test_input/sim_wt/logs/oncotator.log\n",
      "+ oncotator --input_format VCF --log_name test_input/sim_wt/logs/oncotator.log test_input/sim_wt/sim_wt.sorted.vcf test_input/sim_wt/sim_wt.sorted.maf hg19\n",
      "2017-04-14 19:13:28,272 INFO [oncotator.Oncotator:239] Oncotator v1.9.0.0\n",
      "2017-04-14 19:13:28,272 INFO [oncotator.Oncotator:240] Args: Namespace(allow_overwriting=False, cache_url=None, canonical_tx_file=None, collapse_filter_cols=False, collapse_number_annotations=False, dbDir='/xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/', default_cli=[], default_config=None, genome_build='hg19', infer_genotypes='false', infer_onps=False, input_file='test_input/sim_wt/sim_wt.sorted.vcf', input_format='VCF', log_name='test_input/sim_wt/logs/oncotator.log', noMulticore=False, output_file='test_input/sim_wt/sim_wt.sorted.maf', output_format='TCGAMAF', override_cli=[], override_config=None, prepend=False, read_only_cache=False, reannotate_tcga_maf_cols=False, skip_no_alt=False, tx_mode='CANONICAL', verbose=5)\n",
      "2017-04-14 19:13:28,272 INFO [oncotator.Oncotator:241] Log file: /xchip/cga_home/moorena/projects/r2d2/test_input/sim_wt/logs/oncotator.log\n",
      "2017-04-14 19:13:28,275 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cosmic_fusion/hg19/cosmic_fusion.config\n",
      "2017-04-14 19:13:28,275 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/gencode_xrefseq/hg19/gencode_xrefseq.config\n",
      "2017-04-14 19:13:28,275 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/clinvar/hg19/clinvar.config\n",
      "2017-04-14 19:13:28,276 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/esp6500SI_v2/hg19/esp6500SI_v2.config\n",
      "2017-04-14 19:13:28,276 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ccle_by_gene/hg19/ccle_by_gene.config\n",
      "2017-04-14 19:13:28,276 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/oreganno/hg19/oreganno.config\n",
      "2017-04-14 19:13:28,276 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/achilles/hg19/achilles.config\n",
      "2017-04-14 19:13:28,277 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cancer_gene_census/hg19/cancer_gene_census.config\n",
      "2017-04-14 19:13:28,277 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/simple_uniprot/hg19/simple_uniprot.config\n",
      "2017-04-14 19:13:28,277 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/dna_repair_genes/hg19/dna_repair_genes.config\n",
      "2017-04-14 19:13:28,277 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/dbsnp/hg19/dbsnp.config\n",
      "2017-04-14 19:13:28,278 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ref_hg/hg19/ref_hg.config\n",
      "2017-04-14 19:13:28,278 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/uniprot_aa_xform/hg19/uniprot_aa_xform.config\n",
      "2017-04-14 19:13:28,278 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/gencode_out2/hg19/gencode_out2.config\n",
      "2017-04-14 19:13:28,278 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/hgnc/hg19/hgnc.config\n",
      "2017-04-14 19:13:28,279 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ccle_by_gp/hg19/ccle_by_gp.config\n",
      "2017-04-14 19:13:28,279 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cosmic_tissue/hg19/cosmic_tissue.config\n",
      "2017-04-14 19:13:28,279 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/familial/hg19/familial.config\n",
      "2017-04-14 19:13:28,279 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/tumorscape/hg19/tumorscape.config\n",
      "2017-04-14 19:13:28,280 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cosmic/hg19/cosmic.config\n",
      "2017-04-14 19:13:28,280 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/1000genome_db/hg19/1000genome_db.config\n",
      "2017-04-14 19:13:28,280 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/uniprot_aa_annotation/hg19/uniprot_aa_annotation.config\n",
      "2017-04-14 19:13:28,280 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/dbNSFP_ds/hg19/dbNSFP_ds.config\n",
      "2017-04-14 19:13:28,281 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/esp6500SI_v2_coverage/hg19/esp6500SI_v2_coverage.config\n",
      "2017-04-14 19:13:28,281 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/so_terms/hg19/so_terms.config\n",
      "2017-04-14 19:13:28,281 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/tcgascape/hg19/tcgascape.config\n",
      "2017-04-14 19:13:28,281 INFO [oncotator.DatasourceFactory:286] Queuing datasource creation for /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/mutsig/hg19/mutsig.config\n",
      "2017-04-14 19:13:28,281 INFO [oncotator.DatasourceFactory:304] Creating pool\n",
      "2017-04-14 19:13:28,293 INFO [oncotator.DatasourceFactory:307] Pool created\n",
      "2017-04-14 19:13:28,373 INFO [oncotator.datasources.EnsemblTranscriptDatasource:95] GENCODE v19 is being set up in faster, NOT thread-safe mode (for annotation).  \n",
      "2017-04-14 19:13:28,453 INFO [oncotator.datasources.EnsemblTranscriptDatasource:106] GENCODE v19 is being set up with default tx-mode: CANONICAL.  \n",
      "2017-04-14 19:13:28,453 WARNING [oncotator.datasources.EnsemblTranscriptDatasource:123] Attempting to set transcript mode of CANONICAL for ensembl.  This operation is only supported for GENCODE.  Otherwise, will be the same as EFFECT.\n",
      "2017-04-14 19:13:28,453 INFO [oncotator.datasources.EnsemblTranscriptDatasource:109] GENCODE v19 is being set up with basic filtering.  \n",
      "2017-04-14 19:13:28,651 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/so_terms/hg19//SO_Ensembl_consequence_terms.bin...\n",
      "2017-04-14 19:13:28,664 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/oreganno/hg19//oreganno_results_by_pos.hg19.import.bin...\n",
      "2017-04-14 19:13:28,664 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cosmic_fusion/hg19//CosmicMutantExportIncFus_v62_291112_fusion.bin...\n",
      "2017-04-14 19:13:28,664 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/clinvar/hg19//clinvar_hgmd.12-03-20.bin...\n",
      "2017-04-14 19:13:28,664 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cancer_gene_census/hg19//CancerGeneCensus_Table_1_full_2012-03-15.bin...\n",
      "2017-04-14 19:13:28,668 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/gencode_xrefseq/hg19//gencode_xrefseq_v19.bin...\n",
      "2017-04-14 19:13:28,679 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/simple_uniprot/hg19//simple_uniprot_Dec012014.bin...\n",
      "2017-04-14 19:13:29,191 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/dna_repair_genes/hg19//Wood_DNARepairGenes.no_dupes.bin...\n",
      "2017-04-14 19:13:29,196 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/hgnc/hg19//hgnc_download_Sept172014.bin...\n",
      "2017-04-14 19:13:29,262 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/achilles/hg19//achilles_lineage_results.import.bin...\n",
      "2017-04-14 19:13:29,374 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ccle_by_gp/hg19//ccle_results_by_pos.hg19.import.bin...\n",
      "2017-04-14 19:13:29,384 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/cosmic_tissue/hg19//CosmicMutantExportIncFus_v62_291112.geneHistology.bin...\n",
      "2017-04-14 19:13:29,527 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/familial/hg19//Familial Cancer Genes.no_dupes.bin...\n",
      "2017-04-14 19:13:29,534 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/tumorscape/hg19//tumorscape_20100104_generated.bin...\n",
      "2017-04-14 19:13:29,707 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/uniprot_aa_annotation/hg19//uniprot_aa_annotations.bin...\n",
      "2017-04-14 19:13:29,764 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/mutsig/hg19//mutsig_results.import.20110905.bin...\n",
      "2017-04-14 19:13:30,627 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/ccle_by_gene/hg19//ccle_results_by_gene.import.bin...\n",
      "2017-04-14 19:13:33,863 INFO [root:73] Loading /xchip/cga/reference/annotation/db/oncotator_v1_ds_gencode_current/tcgascape/hg19//tcgascape_110405_generated.bin...\n",
      "2017-04-14 19:13:39,226 INFO [oncotator.DatasourceFactory:321] Mapping complete: 17 datasources created in multiprocess\n",
      "2017-04-14 19:13:39,601 INFO [oncotator.utils.RunSpecificationFactory:124] Setting GENCODE v19 to tx-mode of CANONICAL...\n",
      "2017-04-14 19:13:39,602 WARNING [oncotator.datasources.EnsemblTranscriptDatasource:123] Attempting to set transcript mode of CANONICAL for ensembl.  This operation is only supported for GENCODE.  Otherwise, will be the same as EFFECT.\n",
      "2017-04-14 19:13:39,602 INFO [oncotator.utils.RunSpecificationFactory:135] No custom canonical transcripts specified.\n",
      "2017-04-14 19:13:39,604 INFO [oncotator.utils.ConfigUtils:197] Could not find config file (tcgaMAF2.4_output.config).  Trying configs/ prepend.\n",
      "2017-04-14 19:13:39,612 INFO [oncotator.output.TcgaMafOutputRenderer:93] Building alternative keys dictionary...\n",
      "2017-04-14 19:13:39,649 INFO [oncotator.cache.DummyCache:57] No cache specified.  All cache attempts will be listed as cache misses.\n",
      "2017-04-14 19:13:39,649 INFO [oncotator.Annotator:426] Annotating with 27 datasources: Oncotator v1.9.0.0 | Flat File Reference hg19 | GENCODE v19 CANONICAL | UniProt_AAxform 2014_12 | ESP 6500SI-V2 | dbSNP build 134 | COSMIC v62_291112 | 1000Genome phase1 | dbNSFP v2.4 | ESP 6500SI-V2 | ClinVar 12.03.20 | ORegAnno UCSC Track | CCLE_By_GP 09292010 | UniProt_AA 2014_12 | Ensembl ICGC MUCOPA | COSMIC_FusionGenes v62_291112 | gencode_xref_refseq metadata_v19 | CCLE_By_Gene 09292010 | ACHILLES_Lineage_Results 110303 | CGC full_2012-03-15 | UniProt 2014_12 | HumanDNARepairGenes 20110905 | HGNC Sept172014 | COSMIC_Tissue 291112 | Familial_Cancer_Genes 20110905 | TUMORScape 20100104 | TCGAScape 110405 | MutSig Published Results 20110905 \n",
      "2017-04-14 19:13:39,650 INFO [oncotator.utils.ConfigUtils:197] Could not find config file (vcf.in.config).  Trying configs/ prepend.\n",
      "2017-04-14 19:13:39,670 INFO [oncotator.output.TcgaMafOutputRenderer:256] TCGA MAF output file: test_input/sim_wt/sim_wt.sorted.maf\n",
      "2017-04-14 19:13:39,671 INFO [oncotator.output.TcgaMafOutputRenderer:257] Render starting...\n",
      "2017-04-14 19:13:39,671 INFO [oncotator.utils.ConfigUtils:197] Could not find config file (vcf.in.config).  Trying configs/ prepend.\n",
      "2017-04-14 19:13:39,693 INFO [oncotator.output.TcgaMafOutputRenderer:342] Rendered all 0 mutations.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "eval `/broad/software/dotkit/init -b`\n",
    "reuse -q .python-2.7.1-sqlite3-rtrees\n",
    "use -q Oncotator\n",
    "\n",
    "declare -A vcfs=(\n",
    "    [\"wt\"]=\"test_input/sim_wt/sim_wt.sorted.vcf\" \\\n",
    "    [\"het\"]=\"test_input/sim_het/sim_het.sorted.vcf\" \\\n",
    "    [\"hom\"]=\"test_input/sim_hom/sim_hom.sorted.vcf\" \\\n",
    ")\n",
    "\n",
    "declare -A outputs=(\n",
    "    [\"wt\"]=\"test_input/sim_wt/sim_wt.sorted.maf\" \\\n",
    "    [\"het\"]=\"test_input/sim_het/sim_het.sorted.maf\" \\\n",
    "    [\"hom\"]=\"test_input/sim_hom/sim_hom.sorted.maf\" \\\n",
    ")\n",
    "\n",
    "set -euxo pipefail\n",
    "\n",
    "for type in \"${!vcfs[@]}\"; do\n",
    "    log_loc=\"$(dirname ${vcfs[$type]})/logs/oncotator.log\"\n",
    "    oncotator --input_format VCF --log_name \"$log_loc\" \"${vcfs[$type]}\" \"${outputs[$type]}\" hg19\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create symlinks for each possible mutation scenario.\n",
    "\n",
    "Scenarios stored in `test_input/scenarios`.\n",
    "\n",
    "| Scen.\\Files | dna_normal.maf     | dna_tumor.maf      | rna_normal.maf     | rna_tumor.maf      |\n",
    "|-------------|--------------------|--------------------|--------------------|--------------------|\n",
    "| RNAed       | sim_wt.sorted.maf  | sim_wt.sorted.maf  | sim_het.sorted.maf | sim_het.sorted.maf |\n",
    "| T-RNAed     | sim_wt_sorted.maf  | sim_wt_sorted.maf  | sim_wt.sorted.maf  | sim_het.sorted.maf |\n",
    "| VSE         | sim_het.sorted.maf | sim_het.sorted.maf | sim_hom.sorted.maf | sim_hom.sorted.maf |\n",
    "| T-VSE       | sim_het.sorted.maf | sim_het.sorted.maf | sim_het.sorted.maf | sim_hom.sorted.maf |\n",
    "| VSL         | sim_het.sorted.maf | sim_het.sorted.maf | sim_wt.sorted.maf  | sim_wt.sorted.maf  |\n",
    "| T-VSL       | sim_het.sorted.maf | sim_het.sorted.maf | sim_het.sorted.maf | sim_wt.sorted.maf  |\n",
    "| LOH (amp)   | sim_het.sorted.maf | sim_hom.sorted.maf | sim_het.sorted.maf | sim_hom.sorted.maf |\n",
    "| LOH (del)   | sim_het.sorted.maf | sim_wt.sorted.maf  | sim_het.sorted.maf | sim_wt.sorted.maf  |\n",
    "| SOM         | sim_wt.sorted.maf  | sim_het.sorted.maf | sim_wt.sorted.maf  | sim_het.sorted.maf |"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "+ for scen in '\"${!scenarios[@]}\"'\n",
      "+ scen_types=(${scenarios[$scen]})\n",
      "++ seq 0 3\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_wt/sim_wt.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/t-rnaed\n",
      "+ ln -sf ../../sim_wt/sim_wt.sorted.maf test_input/scenarios/t-rnaed/dna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_wt/sim_wt.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/t-rnaed\n",
      "+ ln -sf ../../sim_wt/sim_wt.sorted.maf test_input/scenarios/t-rnaed/dna_tumor.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_wt/sim_wt.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/t-rnaed\n",
      "+ ln -sf ../../sim_wt/sim_wt.sorted.maf test_input/scenarios/t-rnaed/rna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/t-rnaed\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/t-rnaed/rna_tumor.maf\n",
      "+ for scen in '\"${!scenarios[@]}\"'\n",
      "+ scen_types=(${scenarios[$scen]})\n",
      "++ seq 0 3\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/vse\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/vse/dna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/vse\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/vse/dna_tumor.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_hom/sim_hom.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/vse\n",
      "+ ln -sf ../../sim_hom/sim_hom.sorted.maf test_input/scenarios/vse/rna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_hom/sim_hom.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/vse\n",
      "+ ln -sf ../../sim_hom/sim_hom.sorted.maf test_input/scenarios/vse/rna_tumor.maf\n",
      "+ for scen in '\"${!scenarios[@]}\"'\n",
      "+ scen_types=(${scenarios[$scen]})\n",
      "++ seq 0 3\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_wt/sim_wt.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/rnaed\n",
      "+ ln -sf ../../sim_wt/sim_wt.sorted.maf test_input/scenarios/rnaed/dna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_wt/sim_wt.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/rnaed\n",
      "+ ln -sf ../../sim_wt/sim_wt.sorted.maf test_input/scenarios/rnaed/dna_tumor.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/rnaed\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/rnaed/rna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/rnaed\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/rnaed/rna_tumor.maf\n",
      "+ for scen in '\"${!scenarios[@]}\"'\n",
      "+ scen_types=(${scenarios[$scen]})\n",
      "++ seq 0 3\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/vsl\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/vsl/dna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/vsl\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/vsl/dna_tumor.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_wt/sim_wt.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/vsl\n",
      "+ ln -sf ../../sim_wt/sim_wt.sorted.maf test_input/scenarios/vsl/rna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_wt/sim_wt.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/vsl\n",
      "+ ln -sf ../../sim_wt/sim_wt.sorted.maf test_input/scenarios/vsl/rna_tumor.maf\n",
      "+ for scen in '\"${!scenarios[@]}\"'\n",
      "+ scen_types=(${scenarios[$scen]})\n",
      "++ seq 0 3\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/t-vse\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/t-vse/dna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/t-vse\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/t-vse/dna_tumor.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/t-vse\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/t-vse/rna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_hom/sim_hom.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/t-vse\n",
      "+ ln -sf ../../sim_hom/sim_hom.sorted.maf test_input/scenarios/t-vse/rna_tumor.maf\n",
      "+ for scen in '\"${!scenarios[@]}\"'\n",
      "+ scen_types=(${scenarios[$scen]})\n",
      "++ seq 0 3\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/t-vsl\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/t-vsl/dna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/t-vsl\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/t-vsl/dna_tumor.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/t-vsl\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/t-vsl/rna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_wt/sim_wt.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/t-vsl\n",
      "+ ln -sf ../../sim_wt/sim_wt.sorted.maf test_input/scenarios/t-vsl/rna_tumor.maf\n",
      "+ for scen in '\"${!scenarios[@]}\"'\n",
      "+ scen_types=(${scenarios[$scen]})\n",
      "++ seq 0 3\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/loh-amp\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/loh-amp/dna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_hom/sim_hom.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/loh-amp\n",
      "+ ln -sf ../../sim_hom/sim_hom.sorted.maf test_input/scenarios/loh-amp/dna_tumor.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/loh-amp\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/loh-amp/rna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_hom/sim_hom.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/loh-amp\n",
      "+ ln -sf ../../sim_hom/sim_hom.sorted.maf test_input/scenarios/loh-amp/rna_tumor.maf\n",
      "+ for scen in '\"${!scenarios[@]}\"'\n",
      "+ scen_types=(${scenarios[$scen]})\n",
      "++ seq 0 3\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/loh-del\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/loh-del/dna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_wt/sim_wt.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/loh-del\n",
      "+ ln -sf ../../sim_wt/sim_wt.sorted.maf test_input/scenarios/loh-del/dna_tumor.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/loh-del\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/loh-del/rna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_wt/sim_wt.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/loh-del\n",
      "+ ln -sf ../../sim_wt/sim_wt.sorted.maf test_input/scenarios/loh-del/rna_tumor.maf\n",
      "+ for scen in '\"${!scenarios[@]}\"'\n",
      "+ scen_types=(${scenarios[$scen]})\n",
      "++ seq 0 3\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_wt/sim_wt.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/som\n",
      "+ ln -sf ../../sim_wt/sim_wt.sorted.maf test_input/scenarios/som/dna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/som\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/som/dna_tumor.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_wt/sim_wt.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/som\n",
      "+ ln -sf ../../sim_wt/sim_wt.sorted.maf test_input/scenarios/som/rna_normal.maf\n",
      "+ for i in '$(seq 0 $((${#output_files[@]}-1)))'\n",
      "+ case \"${scen_types[$i]}\" in\n",
      "+ target_bam=../../sim_het/sim_het.sorted.maf\n",
      "+ mkdir -p test_input/scenarios/som\n",
      "+ ln -sf ../../sim_het/sim_het.sorted.maf test_input/scenarios/som/rna_tumor.maf\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "declare -A scenarios=(\\\n",
    "    [\"rnaed\"]=\"wt wt het het\" \\\n",
    "    [\"t-rnaed\"]=\"wt wt wt het\" \\\n",
    "    [\"vse\"]=\"het het hom hom\" \\\n",
    "    [\"t-vse\"]=\"het het het hom\" \\\n",
    "    [\"vsl\"]=\"het het wt wt\" \\\n",
    "    [\"t-vsl\"]=\"het het het wt\" \\\n",
    "    [\"loh-amp\"]=\"het hom het hom\" \\\n",
    "    [\"loh-del\"]=\"het wt het wt\" \\\n",
    "    [\"som\"]=\"wt het wt het\" \\\n",
    ")\n",
    "\n",
    "output_files=(\"dna_normal.maf\" \"dna_tumor.maf\" \"rna_normal.maf\" \"rna_tumor.maf\")\n",
    "\n",
    "# BAMs (target paths to ln) must be relative to the directories in which the symlinks are created\n",
    "wt_maf=\"../../sim_wt/sim_wt.sorted.maf\"\n",
    "het_maf=\"../../sim_het/sim_het.sorted.maf\"\n",
    "hom_maf=\"../../sim_hom/sim_hom.sorted.maf\"\n",
    "scen_dir=\"test_input/scenarios\"\n",
    "\n",
    "set -euxo pipefail\n",
    "\n",
    "for scen in \"${!scenarios[@]}\"; do\n",
    "    scen_types=(${scenarios[$scen]})\n",
    "    for i in $(seq 0 $((${#output_files[@]}-1))); do\n",
    "        case \"${scen_types[$i]}\" in\n",
    "            \"wt\") target_bam=\"$wt_maf\";;\n",
    "            \"het\") target_bam=\"$het_maf\";;\n",
    "            \"hom\") target_bam=\"$hom_maf\";;\n",
    "        esac\n",
    "\n",
    "        mkdir -p \"$scen_dir/$scen\"\n",
    "        ln -sf \"$target_bam\" \"$scen_dir/$scen/${output_files[$i]}\"\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rnaed: rnaed_all_inputs\n",
      "t-rnaed: t_rnaed_all_inputs\n",
      "vse: vse_all_inputs\n",
      "t-vse: vse_all_inputs\n",
      "vsl: \n",
      "t-vsl: t_vsl_all_inputs\n",
      "loh-amp: loh_all_inputs\n",
      "loh-del: loh_all_inputs\n",
      "som: somatic_all_inputs\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "+ for scen in '\"${scenarios[@]}\"'\n",
      "+ output_path=test_input/output/rnaed.tsv\n",
      "+ python r2d2.py --dna_normal test_input/scenarios//rnaed/dna_normal.maf --dna_tumor test_input/scenarios//rnaed/dna_tumor.maf --rna_normal test_input/scenarios//rnaed/rna_normal.maf --rna_tumor test_input/scenarios//rnaed/rna_tumor.maf --output test_input/output/rnaed.tsv\n",
      "/home/unix/moorena/.conda/envs/general/lib/python2.7/site-packages/pandas/tools/merge.py:1376: RuntimeWarning: divide by zero encountered in long_scalars\n",
      "  stride //= shape[i]\n",
      "++ cat test_input/output/rnaed.tsv\n",
      "++ awk '{if(NR==2) print $1}'\n",
      "+ echo -e 'rnaed: rnaed_all_inputs'\n",
      "+ for scen in '\"${scenarios[@]}\"'\n",
      "+ output_path=test_input/output/t-rnaed.tsv\n",
      "+ python r2d2.py --dna_normal test_input/scenarios//t-rnaed/dna_normal.maf --dna_tumor test_input/scenarios//t-rnaed/dna_tumor.maf --rna_normal test_input/scenarios//t-rnaed/rna_normal.maf --rna_tumor test_input/scenarios//t-rnaed/rna_tumor.maf --output test_input/output/t-rnaed.tsv\n",
      "/home/unix/moorena/.conda/envs/general/lib/python2.7/site-packages/pandas/tools/merge.py:1376: RuntimeWarning: divide by zero encountered in long_scalars\n",
      "  stride //= shape[i]\n",
      "++ cat test_input/output/t-rnaed.tsv\n",
      "++ awk '{if(NR==2) print $1}'\n",
      "+ echo -e 't-rnaed: t_rnaed_all_inputs'\n",
      "+ for scen in '\"${scenarios[@]}\"'\n",
      "+ output_path=test_input/output/vse.tsv\n",
      "+ python r2d2.py --dna_normal test_input/scenarios//vse/dna_normal.maf --dna_tumor test_input/scenarios//vse/dna_tumor.maf --rna_normal test_input/scenarios//vse/rna_normal.maf --rna_tumor test_input/scenarios//vse/rna_tumor.maf --output test_input/output/vse.tsv\n",
      "++ cat test_input/output/vse.tsv\n",
      "++ awk '{if(NR==2) print $1}'\n",
      "+ echo -e 'vse: vse_all_inputs'\n",
      "+ for scen in '\"${scenarios[@]}\"'\n",
      "+ output_path=test_input/output/t-vse.tsv\n",
      "+ python r2d2.py --dna_normal test_input/scenarios//t-vse/dna_normal.maf --dna_tumor test_input/scenarios//t-vse/dna_tumor.maf --rna_normal test_input/scenarios//t-vse/rna_normal.maf --rna_tumor test_input/scenarios//t-vse/rna_tumor.maf --output test_input/output/t-vse.tsv\n",
      "++ cat test_input/output/t-vse.tsv\n",
      "++ awk '{if(NR==2) print $1}'\n",
      "+ echo -e 't-vse: vse_all_inputs'\n",
      "+ for scen in '\"${scenarios[@]}\"'\n",
      "+ output_path=test_input/output/vsl.tsv\n",
      "+ python r2d2.py --dna_normal test_input/scenarios//vsl/dna_normal.maf --dna_tumor test_input/scenarios//vsl/dna_tumor.maf --rna_normal test_input/scenarios//vsl/rna_normal.maf --rna_tumor test_input/scenarios//vsl/rna_tumor.maf --output test_input/output/vsl.tsv\n",
      "++ cat test_input/output/vsl.tsv\n",
      "++ awk '{if(NR==2) print $1}'\n",
      "+ echo -e 'vsl: '\n",
      "+ for scen in '\"${scenarios[@]}\"'\n",
      "+ output_path=test_input/output/t-vsl.tsv\n",
      "+ python r2d2.py --dna_normal test_input/scenarios//t-vsl/dna_normal.maf --dna_tumor test_input/scenarios//t-vsl/dna_tumor.maf --rna_normal test_input/scenarios//t-vsl/rna_normal.maf --rna_tumor test_input/scenarios//t-vsl/rna_tumor.maf --output test_input/output/t-vsl.tsv\n",
      "++ cat test_input/output/t-vsl.tsv\n",
      "++ awk '{if(NR==2) print $1}'\n",
      "+ echo -e 't-vsl: t_vsl_all_inputs'\n",
      "+ for scen in '\"${scenarios[@]}\"'\n",
      "+ output_path=test_input/output/loh-amp.tsv\n",
      "+ python r2d2.py --dna_normal test_input/scenarios//loh-amp/dna_normal.maf --dna_tumor test_input/scenarios//loh-amp/dna_tumor.maf --rna_normal test_input/scenarios//loh-amp/rna_normal.maf --rna_tumor test_input/scenarios//loh-amp/rna_tumor.maf --output test_input/output/loh-amp.tsv\n",
      "++ cat test_input/output/loh-amp.tsv\n",
      "++ awk '{if(NR==2) print $1}'\n",
      "+ echo -e 'loh-amp: loh_all_inputs'\n",
      "+ for scen in '\"${scenarios[@]}\"'\n",
      "+ output_path=test_input/output/loh-del.tsv\n",
      "+ python r2d2.py --dna_normal test_input/scenarios//loh-del/dna_normal.maf --dna_tumor test_input/scenarios//loh-del/dna_tumor.maf --rna_normal test_input/scenarios//loh-del/rna_normal.maf --rna_tumor test_input/scenarios//loh-del/rna_tumor.maf --output test_input/output/loh-del.tsv\n",
      "++ cat test_input/output/loh-del.tsv\n",
      "++ awk '{if(NR==2) print $1}'\n",
      "+ echo -e 'loh-del: loh_all_inputs'\n",
      "+ for scen in '\"${scenarios[@]}\"'\n",
      "+ output_path=test_input/output/som.tsv\n",
      "+ python r2d2.py --dna_normal test_input/scenarios//som/dna_normal.maf --dna_tumor test_input/scenarios//som/dna_tumor.maf --rna_normal test_input/scenarios//som/rna_normal.maf --rna_tumor test_input/scenarios//som/rna_tumor.maf --output test_input/output/som.tsv\n",
      "++ cat test_input/output/som.tsv\n",
      "++ awk '{if(NR==2) print $1}'\n",
      "+ echo -e 'som: somatic_all_inputs'\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "scenarios=(\"rnaed\" \"t-rnaed\" \"vse\" \"t-vse\" \"vsl\" \"t-vsl\" \"loh-amp\" \"loh-del\" \"som\")\n",
    "base_dir=\"test_input/scenarios/\"\n",
    "output_dir=\"test_input/output\"\n",
    "mkdir -p \"$output_dir\"\n",
    ". venv/bin/activate\n",
    "\n",
    "set -euxo pipefail\n",
    "\n",
    "for scen in \"${scenarios[@]}\"; do\n",
    "    output_path=\"test_input/output/$scen.tsv\"\n",
    "    python r2d2.py \\\n",
    "        --dna_normal \"$base_dir/$scen/dna_normal.maf\" \\\n",
    "        --dna_tumor \"$base_dir/$scen/dna_tumor.maf\" \\\n",
    "        --rna_normal \"$base_dir/$scen/rna_normal.maf\" \\\n",
    "        --rna_tumor \"$base_dir/$scen/rna_tumor.maf\" \\\n",
    "        --output \"$output_path\"\n",
    "        \n",
    "    echo -e \"$scen: $(cat $output_path | awk '{if(NR==2) print $1}')\"\n",
    "done"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
